### PR TITLE
saxon 12.0

### PR DESCRIPTION
**Pull request by joewiz**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by joewiz** [comment]
As the change to `url` suggests, the Saxon project [announced](https://www.saxonica.com/about/news.xml) on January 12:

> The Saxon open-source project at SourceForge is now obsolete. From now on open-source releases will be available on GitHub instead. The Saxon-HE repository contains current releases, while the Saxon-Archive repository contains older releases and other content migrated from SourceForge.

Thus I think the `homepage` should also be updated from https://saxon.sourceforge.io to https://github.com/Saxonica/Saxon-HE/tree/main/12 - the homepage of the SaxonHE edition of Saxon. Alternatively, it could be the official homepage of Saxon, https://www.saxonica.com/, or the page on saxonica.com that links to GitHub, https://www.saxonica.com/download/java.xml. I'm open to any of these and would welcome advice from Homebrew users/maintainers about which one would be the most appropriate from a Homebrew perspective.

**Comment by joewiz** [comment]
It appears that the `livecheck` stanza refers to the `homepage` URL in some way (I see no mention of `:stable` on https://docs.brew.sh/Cask-Cookbook#stanza-livecheck)... The page that the Saxon maintainers use to list the current version of the project is https://github.com/Saxonica/Saxon-HE:

> The latest release of Saxon-HE is [12.0](https://github.com/Saxonica/Saxon-HE/blob/main/12).

So I'm also open to advice about whether `livecheck` should rely on `homepage` (presumably via this `:stable` property?), or whether a literal URL should be supplied for livecheck's purposes.

**Comment by joewiz** [comment]
I see the CI checks involving bottles seem to have failed. Could anyone shed light on the underlying cause? Is this PR missing something?

**Comment by chenrui333** [comment]
> As the change to `url` suggests, the Saxon project [announced](https://www.saxonica.com/about/news.xml) on January 12:
> 
> > The Saxon open-source project at SourceForge is now obsolete. From now on open-source releases will be available on GitHub instead. The Saxon-HE repository contains current releases, while the Saxon-Archive repository contains older releases and other content migrated from SourceForge.
> 
> Thus I think the `homepage` should also be updated from https://saxon.sourceforge.io to https://github.com/Saxonica/Saxon-HE/tree/main/12 - the homepage of the SaxonHE edition of Saxon. Alternatively, it could be the official homepage of Saxon, https://www.saxonica.com/, or the page on saxonica.com that links to GitHub, https://www.saxonica.com/download/java.xml. I'm open to any of these and would welcome advice from Homebrew users/maintainers about which one would be the most appropriate from a Homebrew perspective.

why dont they use github releases to host the release artifacts, that is interesting.

**Comment by chenrui333** [comment]
> It appears that the `livecheck` stanza refers to the `homepage` URL in some way (I see no mention of `:stable` on https://docs.brew.sh/Cask-Cookbook#stanza-livecheck)... The page that the Saxon maintainers use to list the current version of the project is https://github.com/Saxonica/Saxon-HE:
> 
> > The latest release of Saxon-HE is [12.0](https://github.com/Saxonica/Saxon-HE/blob/main/12).
> 
> So I'm also open to advice about whether `livecheck` should rely on `homepage` (presumably via this `:stable` property?), or whether a literal URL should be supplied for livecheck's purposes.

livecheck is fine, we can update it.

**Comment by chenrui333** [comment]
From what I can see, the biggest issue is with the resource location.

File: Formula/saxon.rb
Comment (chenrui333) [inline Formula/saxon.rb:3]: this homepage should be also updated.

**Comment by chenrui333** [comment]
also raised a question about the right place for the artifact.

**Comment by chenrui333** [comment]
this PR would be better on hold until upstream release the artifacts inside github releases.

File: Formula/saxon.rb
Comment (joewiz) [reply Formula/saxon.rb:3]: I mentioned some options [in my first post](https://github.com/Homebrew/homebrew-core/pull/120597#issuecomment-1382823968) above:

> I think the `homepage` should also be updated from https://saxon.sourceforge.io/ to https://github.com/Saxonica/Saxon-HE/tree/main/12 - the homepage of the SaxonHE edition of Saxon. Alternatively, it could be the official homepage of Saxon, https://www.saxonica.com/, or the page on saxonica.com that links to GitHub, https://www.saxonica.com/download/java.xml. I'm open to any of these and would welcome advice from Homebrew users/maintainers about which one would be the most appropriate from a Homebrew perspective.

**Comment by joewiz** [comment]
@chenrui333 Thanks for your review and help finding the right path for this PR!

File: Formula/saxon.rb
Comment (chenrui333) [reply Formula/saxon.rb:3]: yeah, once github releases are in, we can have a default github livecheck strategy in place, that might just do the job.

Comment (joewiz) [reply Formula/saxon.rb:3]: Sounds good. In the meantime, I'll update the PR to switch `homepage` from sourceforge to the new GitHub repo's landing page, https://github.com/Saxonica/Saxon-HE. I think that's the best choice for `homepage`.

File: Formula/saxon.rb
Comment (joewiz) [inline Formula/saxon.rb:6]: For me, `openssl dgst -sha256 SaxonHE12-0J.zip` returns:

```
SHA256(SaxonHE12-0J.zip)= c476746275dd5a0de1d203e89c21a249a02efe33350b560c4086cb08b0816be7
```

I wonder why our calculations differ?

File: Formula/saxon.rb
Comment (SMillerDev) [inline Formula/saxon.rb:4]: Otherwise you're downloading the GitHub website
```suggestion
  url "https://github.com/Saxonica/Saxon-HE/blob/main/12/Java/SaxonHE12-0J.zip?raw=true"
```

Comment (joewiz) [reply Formula/saxon.rb]: @SMillerDev Thank you!!

File: Formula/saxon.rb
Comment (joewiz) [reply Formula/saxon.rb:6]: I think my original sha256 was correct. [CI is choking on this](https://github.com/Homebrew/homebrew-core/actions/runs/3931752350/jobs/6723515405) and reports it expected my original value:

```
Downloaded to: /Users/brew/Library/Caches/Homebrew/downloads/144052f22bce4622346f65b795a6283e6c7e3c29d4d9665cc880ad5f15e550f1--SaxonHE12-0J.zip
  SHA256: c476746275dd5a0de1d203e89c21a2[49](https://github.com/Homebrew/homebrew-core/actions/runs/3931752350/jobs/6723515405#step:6:50)a02efe333[50](https://github.com/Homebrew/homebrew-core/actions/runs/3931752350/jobs/6723515405#step:6:51)b560c4086cb08b0816be7
  Warning: Formula reports different sha256: 32a638c2461d94c3cf68b25658bb475cdfc09c17de36f8b97bbd73bab6e3c4e4
```

**Comment by joewiz** [comment]
This PR now passes CI checks. The only issue is adapting `livecheck` to detect new releases. Since @ndw [said he'd investigate using GitHub Releases](https://github.com/Saxonica/Saxon-HE/issues/1#issuecomment-1382849386) (thanks, Norm!), let's see how this goes.

**Comment by joewiz** [comment]
@SMillerDev Thank you for removing the `livecheck` stanza. I think that should allow this PR to be merged (or perhaps squashed, given the number of commits) as is.

Then, once the upstream project decides on their release artifact publishing strategy, an appropriate `livecheck` stanza can be reintroduced to the formula.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
the commits squash would be handled by @BrewTestBot automatically when PR gets merged.

**Comment by joewiz** [comment]
@chenrui333 @SMillerDev Thank you!

**Pull request closed without merging**
